### ISSUE TITLE
Failing with generating services when datasources connector - rest 

### ISSUE THREAD

**Issue by dooddy** [mean-expert-official/loopback-sdk-builder#99]
I don't know, why I have this issue. But I can provide some info. This is my config files

datasources.json

```json
{
  ...
  "corezoid": {
    "name": "corezoid",
    "crud": false,
    "connector": "rest",
    "options": {
      "headers": {
        "accept": "application/json",
        "content-type": "application/json"
      },
      "strictSSL": false
    },
    "operations": [
      {
        "template": {
          "method": "POST",
          "url": "https://www.corezoid.com/api/1/json/{path}",
          "query": {
          },
          "responsePath": "$",
          "body": "{body}",
          "options": {
            "useQuerystring": false
          }
        },
        "functions": {
          "run": ["path", "body"]
        }
      }
    ]
  }
}
```
and
model-config.json

```json
{
  "_meta": {
    "sources": [
      "loopback/common/models",
      "loopback/server/models",
      "../common/models",
      "./models"
    ],
    "mixins": [
      "loopback/common/mixins",
      "loopback/server/mixins",
      "../common/mixins",
      "./mixins"
    ]
  },
  ...
  "Core": {
    "dataSource": "corezoid",
    "public": true
  }
}
```

and when I try to build sdk then I have this error

![selection_081](https://cloud.githubusercontent.com/assets/12660249/18009661/eb3c5164-6bb5-11e6-82ea-cb607a6169b1.png)

Maybe I missed something?
But on the other side sdk for angular1 executed without errors

**Comment by jonathan-casarrubias** [comment]
Hi @dooddy, thanks for reporting.. Honestly I haven't verified this type of connector before, so I will need to verify if loopback is providing something different for these models or something is missing.

Please share your model configuration json for Core... 

Looks like there is a param of type string that is missing for that model

**Comment by jonathan-casarrubias** [comment]
OK I have verified the file for that section and the only match methods are used either for

a) to match a method name
b) to match a param arg

I see really difficult to have a missing method name, so I think the best option would be that there is a method in the Core Model that has a param configuration that does not provide the arg property, which make sense because there is a validation missing that assumes the arg property will always be available, therefore it seems consistent with your error log stack saying there is not  match property of undefined.

Since is a really specific issue, could you help me with the following:

open the following file

node_modules/@mean-expert/loopback-sdk-builder/lib/angular2/shared/services/custom/service.ejs

And change the line number 59 from

````js
return (param.http && param.http.source === 'path') || (param.arg.match(/(id|fk|file|container)/));
````

To 


````js
return (param.http && param.http.source === 'path') || (param.arg && param.arg.match(/(id|fk|file|container)/));
````

And try to regenerate the sdk? I believe it should work.. if so I will send a fix ASAP

Cheers
Jon

**Comment by dooddy** [comment]
Thanks for reply.

I tried to change string but the another error occurs
![selection_085](https://cloud.githubusercontent.com/assets/12660249/18012155/f1d975e6-6bc0-11e6-9d2e-664fcfbd1bd8.png)


here is my model configuration for core

core.json
```json
{
  "name": "Core",
  "plural": "core",
  "base": "Model",
  "idInjection": true,
  "options": {
    "validateUpsert": true
  },
  "properties": {},
  "validations": [],
  "relations": {},
  "acls": [
    {
      "accessType": "*",
      "property": "smsStatus",
      "principalType": "ROLE",
      "principalId": "$unauthenticated",
      "permission": "ALLOW"
    },
    {
    "accessType": "*",
    "principalType": "ROLE",
    "principalId": "$unauthenticated",
    "permission": "DENY"
  }],
  "methods": {}
}
```

**Comment by jonathan-casarrubias** [comment]
@dooddy thanks!! it actually confirms the issue, is just that there are multiple places where that validation is made, now the issue is in another file but it just confirm that the REST Datsource is creating methods with param properties that does not provide an arg param.

So I will add the validation to check the arg is present in the different places that is needed

**Comment by jonathan-casarrubias** [comment]
@dooddy are you using the invoke method or only the run?... I setup the tests with a rest datasource and got to replicate your issue, once I changed the validations the SDK is now generating that services, but as I imagined the invoke one is corrupted.

So, verifying the API Explorer I see it actually is an application/x-www-form-urlencoded request, so it may be possible that this specific method needs a different handling than the others. This is actually the one that is breaking, does the angular 1 sdk generate the invoke method correctly? it does use the right headers?
